core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -318,7 +318,7 @@ private[spark] class SparkSubmit extends Logging {
 
       if (!StringUtils.isBlank(resolvedMavenCoordinates)) {
         args.jars = mergeFileLists(args.jars, resolvedMavenCoordinates)
-        if (args.isPython) {
+        if (args.isPython || isInternal(args.primaryResource)) {
```

Comment:
These will be dumb questions since I don't know this code, but is this specific just to Python? This seems to set pyFiles even when not running a Python job or am I missing something?

Reply:
The code snippet in the if() statement is to properly set "spark.submit.pyFiles" config, but it only get executed when args.isPython. SparkSubmit uses the primary resource's suffix to determine args.isPython. For NO_RESOURCE type ("spark-internal" as primary resource), we should also do this because it might use python in the NO_RESOURCE scenario.

Reply:
Yeah I get what the code does, was just wondering why it always sets a pyfiles now even when it's not a pyspark app. But the answer is that pyspark apps also need resolved Maven dependencies, I believe. @vanzin does this look right?